File: python/leetcode/157.py
class Solution(object):
    def read(self, buf, n):
        res = 0
        buffer = [''] * 4
        for i in xrange(n / 4 + 1):
            size = read4(buffer)
            if size == 0: break 
            buf[res:res+size] = buffer
            res += size
        return min(res, n)

File: cpp/leetcode/157.cpp
class Solution {
public:
    int read(char *buf, int n) {
        int idx = 0;
        while(idx < n){
            int size = read4(buf + idx);
            idx += size;
            if(size < 4) break;
        }
        return min(idx, n);
    }
};